lib/streamlit/navigation/page.py

```diff
@@ -119,6 +119,11 @@ def __init__(
         if isinstance(page, Path):
             page = (main_path / page).resolve()
 
+            if not page.is_file():
+                raise StreamlitAPIException(
+                    f"The file `{page.name}` does not seem to exist."
```

Comment:
nit: maybe its better to show the full file path here to make it easier for the developer to troubleshoot this?

Reply:
I did this initially but changed it to just the path's basename because I wanted to avoid leaking information about a server's directory structure to the client.

It should be hard for this to be a problem in practice as you'd expect this error to really only be thrown when an app is being developed and not for apps in production, but I think it's worth being safe since you probably don't lose much troubleshooting ability only displaying a file's basename.

Reply:
Maybe something like `Unable to create Page. The file `{page.name}` does not exist.` ? Unless you see a reason to say "seem to".

Reply:
The "seem to" wording is due to the fact that a file may exist, but the process may not have the permissions to access it. I think the distinction isn't important enough to really matter, though, so I'll change this to the suggested wording.

Reply:
Cool - I don't feel strongly, "could not be found" or something like that works too

Reply:
Ah, yeah I think could not be found works for both cases. Will use that